From 1bf7b33b6e944f99cb2ea1a10704300ac6b2d9cc Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Mon, 16 May 2016 13:03:15 -0400
Subject: [PATCH 0445/1239] thunderx: Update configs for DM USB

This commit updates Thunderx 81xx configs for USB device model

Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
---
 configs/thunderx_81xx_defconfig |  4 ++++
 include/configs/thunderx_81xx.h | 14 +++++++++++---
 2 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/configs/thunderx_81xx_defconfig b/configs/thunderx_81xx_defconfig
index 78eaad168e..345844969b 100644
--- a/configs/thunderx_81xx_defconfig
+++ b/configs/thunderx_81xx_defconfig
@@ -28,3 +28,7 @@ CONFIG_DEBUG_UART_PL011=y
 CONFIG_DEBUG_UART_BASE=0x87e028000000
 CONFIG_DEBUG_UART_CLOCK=24000000
 CONFIG_DEBUG_UART_SKIP_INIT=y
+CONFIG_USB=y
+CONFIG_DM_USB=y
+CONFIG_USB_XHCI_HCD=y
+CONFIG_USB_STORAGE=y
diff --git a/include/configs/thunderx_81xx.h b/include/configs/thunderx_81xx.h
index 7cb7088fa4..cf43069e49 100644
--- a/include/configs/thunderx_81xx.h
+++ b/include/configs/thunderx_81xx.h
@@ -7,6 +7,8 @@
 #ifndef __THUNDERX_81XX_H__
 #define __THUNDERX_81XX_H__
 
+#define DEBUG
+
 #define CONFIG_SPECIAL_SYNC_HANDLER
 
 /*#define CONFIG_ARMV8_SWITCH_TO_EL1*/
@@ -154,6 +156,8 @@
 #define CONFIG_MAX_BGX			2
 
 #define CONFIG_PHYLIB
+#define CONFIG_PHYLIB_10G
+
 
 /* Command line configuration */
 #define CONFIG_MENU
@@ -182,6 +186,8 @@
 #define CONFIG_CMD_GREPENV
 #define CONFIG_CMD_ENV_CALLBACK
 
+#define CONFIG_ENV_CALLBACK_LIST_STATIC "txsmi\\d?mode:smimode"
+
 /* AHCI support Definitions */
 #ifdef CONFIG_CMD_SATA
   #define CONFIG_SATA_AHCI
@@ -240,11 +246,9 @@
 #define PHYS_SDRAM_1_SIZE		(0x80000000-MEM_BASE)	/* 2048 MB */
 #define CONFIG_SYS_SDRAM_BASE		PHYS_SDRAM_1
 
-#define CONFIG_USB_XHCI
 #define CONFIG_USB_XHCI_PCI
-#define CONFIG_USB_STORAGE
 #define CONFIG_SYS_USB_XHCI_MAX_ROOT_PORTS 2
-#define CONFIG_USB_MAX_CONTROLLER_COUNT 4
+
 
 
 /* PCIe network controller drivers */
@@ -254,6 +258,8 @@
 
 #define E1000_DEBUG
 
+#define CONFIG_RTL8169
+
 /* Initial environment variables */
 #define UBOOT_IMG_HEAD_SIZE		0x40
 /* C80000 - 0x40 */
@@ -261,6 +267,8 @@
 					"kernel_addr=04007ffc0\0"	\
 					"fdt_addr=0x54C00000\0"		\
 					"fdt_high=0x5fffffff\0"		\
+					"smi0mode=0.0.0\0"		\
+					"smi1mode=0.0.0\0"		\
 					"autoload=0\0"
 
 #define CONFIG_BOOTARGS			\
-- 
2.29.0

